simuleval/agents/states.py

```diff
@@ -29,6 +29,7 @@ def reset(self) -> None:
         self.target_finished = False
         self.source_sample_rate = 0
         self.target_sample_rate = 0
+        self.upstream_states = []
```

Comment:
qq: since this is at the level of an agent rather than the pipeline level, doesn't it require the agent accessing `upstream_states` to make some assumptions about the ordering of the upstream states? I think it works if we don't need to reuse the agent in many different places, but it would be modular than originally intended

Reply:
The order is from the furthest to the closest upstream agent states. This PR still assumes linear pipeline, but I should probably consider tree structures